napari/utils/events/event.py

```diff
@@ -828,6 +837,18 @@ def ignore_callback_errors(self, ignore):
             elif isinstance(emitter, EmitterGroup):
                 emitter.ignore_callback_errors_all(ignore)
 
+    def blocker_all(self):
+        """Return an EventBlockerAll to be used in 'with' statements
+
+        Notes
+        -----
+        For example, one could do::
+
+            with emitter.blocker_all():
+                pass  # ..do stuff; no events will be emitted..
+        """
+        return EventBlockerAll(self)
```

Comment:
```suggestion
    @contextmanager
    def blocker_all(self):
        """Return an EventBlockerAll to be used in 'with' statements

        Notes
        -----
        For example, one could do::

            with emitter.blocker_all():
                pass  # ..do stuff; no events will be emitted..
        """
        self.block_all()
        yield
        self.unblock_all()
```
?

Reply:
yeah, I prefer that too.   @sofroniewn added that and I think he was just trying to stick with the patterns elsewhere in that module for consistency sake.  Do you want to stay with the way vispy does it?  or go our own way with it?

Reply:
i was just copying vispy for consistency, i don't have a strong feeling on this one. i'd say we can revisit in the future if we want to change as I don't want to hold up the PR for this, but absolutely fine to revisit

Reply:
i think it's better to be consistent, and I don't think I want to go chopping up their eventblocker just yet.  so let's leave it